.github/workflows/go-build-and-test.yml

```diff
@@ -0,0 +1,25 @@
+name: build-and-test
+
+on:
+  push:
+    branches: [ "main" ]
```

Comment:
We don't need to re-test when pushing to the branch, at this point the code is merged which defeats the purpose of testing in the presubmits  (preventing the merge of faulty code : ) 
```suggestion
```